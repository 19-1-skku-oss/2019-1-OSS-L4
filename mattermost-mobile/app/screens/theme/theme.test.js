// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import ThemeTile from './theme_tile';

import Preferences from 'mattermost-redux/constants/preferences';

import Theme from './theme';

jest.mock('react-intl');

const allowedThemes = [
    {
        type: 'Mattermost',
        sidebarBg: '#145dbf',
        sidebarText: '#ffffff',
        sidebarUnreadText: '#ffffff',
        sidebarTextHoverBg: '#4578bf',
        sidebarTextActiveBorder: '#579eff',
        sidebarTextActiveColor: '#ffffff',
        sidebarHeaderBg: '#1153ab',
        sidebarHeaderTextColor: '#ffffff',
        onlineIndicator: '#06d6a0',
        awayIndicator: '#ffbc42',
        dndIndicator: '#f74343',
        mentionBg: '#ffffff',
        mentionColor: '#145dbf',
        centerChannelBg: '#ffffff',
        centerChannelColor: '#3d3c40',
        newMessageSeparator: '#ff8800',
        linkColor: '#2389d7',
        buttonBg: '#166de0',
        buttonColor: '#ffffff',
        errorTextColor: '#fd5960',
        mentionHighlightBg: '#ffe577',
        mentionHighlightLink: '#166de0',
        codeTheme: 'github',
        key: 'default',
    },
    {
        type: 'Organization',
        sidebarBg: '#2071a7',
        sidebarText: '#ffffff',
        sidebarUnreadText: '#ffffff',
        sidebarTextHoverBg: '#136197',
        sidebarTextActiveBorder: '#7ab0d6',
        sidebarTextActiveColor: '#ffffff',
        sidebarHeaderBg: '#2f81b7',
        sidebarHeaderTextColor: '#ffffff',
        onlineIndicator: '#7dbe00',
        awayIndicator: '#dcbd4e',
        dndIndicator: '#ff6a6a',
        mentionBg: '#fbfbfb',
        mentionColor: '#2071f7',
        centerChannelBg: '#f2f4f8',
        centerChannelColor: '#333333',
        newMessageSeparator: '#ff8800',
        linkColor: '#2f81b7',
        buttonBg: '#1dacfc',
        buttonColor: '#ffffff',
        errorTextColor: '#a94442',
        mentionHighlightBg: '#f3e197',
        mentionHighlightLink: '#2f81b7',
        codeTheme: 'github',
        key: 'organization',
    },
    {
        type: 'Mattermost Dark',
        sidebarBg: '#1b2c3e',
        sidebarText: '#ffffff',
        sidebarUnreadText: '#ffffff',
        sidebarTextHoverBg: '#4a5664',
        sidebarTextActiveBorder: '#66b9a7',
        sidebarTextActiveColor: '#ffffff',
        sidebarHeaderBg: '#1b2c3e',
        sidebarHeaderTextColor: '#ffffff',
        onlineIndicator: '#65dcc8',
        awayIndicator: '#c1b966',
        dndIndicator: '#e81023',
        mentionBg: '#b74a4a',
        mentionColor: '#ffffff',
        centerChannelBg: '#2f3e4e',
        centerChannelColor: '#dddddd',
        newMessageSeparator: '#5de5da',
        linkColor: '#a4ffeb',
        buttonBg: '#4cbba4',
        buttonColor: '#ffffff',
        errorTextColor: '#ff6461',
        mentionHighlightBg: '#984063',
        mentionHighlightLink: '#a4ffeb',
        codeTheme: 'solarized-dark',
        key: 'mattermostDark',
    },
    {
        type: 'Windows Dark',
        sidebarBg: '#171717',
        sidebarText: '#ffffff',
        sidebarUnreadText: '#ffffff',
        sidebarTextHoverBg: '#302e30',
        sidebarTextActiveBorder: '#196caf',
        sidebarTextActiveColor: '#ffffff',
        sidebarHeaderBg: '#1f1f1f',
        sidebarHeaderTextColor: '#ffffff',
        onlineIndicator: '#399fff',
        awayIndicator: '#c1b966',
        dndIndicator: '#e81023',
        mentionBg: '#0177e7',
        mentionColor: '#ffffff',
        centerChannelBg: '#1f1f1f',
        centerChannelColor: '#dddddd',
        newMessageSeparator: '#cc992d',
        linkColor: '#0d93ff',
        buttonBg: '#0177e7',
        buttonColor: '#ffffff',
        errorTextColor: '#ff6461',
        mentionHighlightBg: '#784098',
        mentionHighlightLink: '#a4ffeb',
        codeTheme: 'monokai',
        key: 'windows10',
    },
];

describe('Theme', () => {
    const baseProps = {
        actions: {
            savePreferences: jest.fn(),
        },
        allowedThemes,
        isLandscape: false,
        isTablet: false,
        navigator: {
            setOnNavigatorEvent: jest.fn(),
        },
        teamId: 'test-team',
        theme: Preferences.THEMES.default,
        userId: 'test-user',
    };

    test('should match snapshot', () => {
        const wrapper = shallow(
            <Theme {...baseProps}/>,
        );

        expect(wrapper.getElement()).toMatchSnapshot();
        expect(wrapper.find(ThemeTile)).toHaveLength(4);
    });
});
