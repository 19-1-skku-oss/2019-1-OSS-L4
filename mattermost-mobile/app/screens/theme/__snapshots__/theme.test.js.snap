// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Theme should match snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <Connect(StatusBar) />
  <View
    style={
      Object {
        "backgroundColor": "rgba(61,60,64,0.06)",
        "flex": 1,
        "paddingTop": 35,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderBottomColor": "rgba(61,60,64,0.1)",
          "borderBottomWidth": 1,
          "borderTopColor": "rgba(61,60,64,0.1)",
          "borderTopWidth": 1,
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "center",
          "marginBottom": 30,
        }
      }
    >
      <ThemeTile
        action={[Function]}
        actionValue="default"
        imageSrc={
          Object {
            "testUri": "../../../dist/assets/images/themes/mattermost.png",
          }
        }
        isLandscape={false}
        isTablet={false}
        label={
          <Text>
            Mattermost
          </Text>
        }
        selected={true}
        theme={
          Object {
            "awayIndicator": "#ffbc42",
            "buttonBg": "#166de0",
            "buttonColor": "#ffffff",
            "centerChannelBg": "#ffffff",
            "centerChannelColor": "#3d3c40",
            "codeTheme": "github",
            "dndIndicator": "#f74343",
            "errorTextColor": "#fd5960",
            "linkColor": "#2389d7",
            "mentionBj": "#ffffff",
            "mentionColor": "#145dbf",
            "mentionHighlightBg": "#ffe577",
            "mentionHighlightLink": "#166de0",
            "newMessageSeparator": "#ff8800",
            "onlineIndicator": "#06d6a0",
            "sidebarBg": "#145dbf",
            "sidebarHeaderBg": "#1153ab",
            "sidebarHeaderTextColor": "#ffffff",
            "sidebarText": "#ffffff",
            "sidebarTextActiveBorder": "#579eff",
            "sidebarTextActiveColor": "#ffffff",
            "sidebarTextHoverBg": "#4578bf",
            "sidebarUnreadText": "#ffffff",
            "type": "Mattermost",
          }
        }
      />
      <ThemeTile
        action={[Function]}
        actionValue="organization"
        imageSrc={
          Object {
            "testUri": "../../../dist/assets/images/themes/organization.png",
          }
        }
        isLandscape={false}
        isTablet={false}
        label={
          <Text>
            Organization
          </Text>
        }
        selected={false}
        theme={
          Object {
            "awayIndicator": "#ffbc42",
            "buttonBg": "#166de0",
            "buttonColor": "#ffffff",
            "centerChannelBg": "#ffffff",
            "centerChannelColor": "#3d3c40",
            "codeTheme": "github",
            "dndIndicator": "#f74343",
            "errorTextColor": "#fd5960",
            "linkColor": "#2389d7",
            "mentionBj": "#ffffff",
            "mentionColor": "#145dbf",
            "mentionHighlightBg": "#ffe577",
            "mentionHighlightLink": "#166de0",
            "newMessageSeparator": "#ff8800",
            "onlineIndicator": "#06d6a0",
            "sidebarBg": "#145dbf",
            "sidebarHeaderBg": "#1153ab",
            "sidebarHeaderTextColor": "#ffffff",
            "sidebarText": "#ffffff",
            "sidebarTextActiveBorder": "#579eff",
            "sidebarTextActiveColor": "#ffffff",
            "sidebarTextHoverBg": "#4578bf",
            "sidebarUnreadText": "#ffffff",
            "type": "Mattermost",
          }
        }
      />
      <ThemeTile
        action={[Function]}
        actionValue="mattermostDark"
        imageSrc={
          Object {
            "testUri": "../../../dist/assets/images/themes/mattermost_dark.png",
          }
        }
        isLandscape={false}
        isTablet={false}
        label={
          <Text>
            Mattermost Dark
          </Text>
        }
        selected={false}
        theme={
          Object {
            "awayIndicator": "#ffbc42",
            "buttonBg": "#166de0",
            "buttonColor": "#ffffff",
            "centerChannelBg": "#ffffff",
            "centerChannelColor": "#3d3c40",
            "codeTheme": "github",
            "dndIndicator": "#f74343",
            "errorTextColor": "#fd5960",
            "linkColor": "#2389d7",
            "mentionBj": "#ffffff",
            "mentionColor": "#145dbf",
            "mentionHighlightBg": "#ffe577",
            "mentionHighlightLink": "#166de0",
            "newMessageSeparator": "#ff8800",
            "onlineIndicator": "#06d6a0",
            "sidebarBg": "#145dbf",
            "sidebarHeaderBg": "#1153ab",
            "sidebarHeaderTextColor": "#ffffff",
            "sidebarText": "#ffffff",
            "sidebarTextActiveBorder": "#579eff",
            "sidebarTextActiveColor": "#ffffff",
            "sidebarTextHoverBg": "#4578bf",
            "sidebarUnreadText": "#ffffff",
            "type": "Mattermost",
          }
        }
      />
      <ThemeTile
        action={[Function]}
        actionValue="windows10"
        imageSrc={
          Object {
            "testUri": "../../../dist/assets/images/themes/windows_dark.png",
          }
        }
        isLandscape={false}
        isTablet={false}
        label={
          <Text>
            Windows Dark
          </Text>
        }
        selected={false}
        theme={
          Object {
            "awayIndicator": "#ffbc42",
            "buttonBg": "#166de0",
            "buttonColor": "#ffffff",
            "centerChannelBg": "#ffffff",
            "centerChannelColor": "#3d3c40",
            "codeTheme": "github",
            "dndIndicator": "#f74343",
            "errorTextColor": "#fd5960",
            "linkColor": "#2389d7",
            "mentionBj": "#ffffff",
            "mentionColor": "#145dbf",
            "mentionHighlightBg": "#ffe577",
            "mentionHighlightLink": "#166de0",
            "newMessageSeparator": "#ff8800",
            "onlineIndicator": "#06d6a0",
            "sidebarBg": "#145dbf",
            "sidebarHeaderBg": "#1153ab",
            "sidebarHeaderTextColor": "#ffffff",
            "sidebarText": "#ffffff",
            "sidebarTextActiveBorder": "#579eff",
            "sidebarTextActiveColor": "#ffffff",
            "sidebarTextHoverBg": "#4578bf",
            "sidebarUnreadText": "#ffffff",
            "type": "Mattermost",
          }
        }
      />
    </View>
  </View>
</View>
`;
